pycyphal/transport/udp/_ip/_v4.py

```diff
@@ -27,18 +31,16 @@ class IPv4SocketFactory(SocketFactory):
     a node-ID that maps to the broadcast address for the subnet is unavailable.
     """
 
-    def __init__(self, local_ip_address: IPv4Address):
-        if not isinstance(local_ip_address, IPv4Address):  # pragma: no cover
-            raise TypeError(f"Unexpected IP address type: {type(local_ip_address)}")
-        self._local = local_ip_address
+    def __init__(self, local_ip_addr: typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address]):
+        self._local_ip_addr = local_ip_addr
 
     @property
     def max_nodes(self) -> int:
-        return IP_ADDRESS_NODE_ID_MASK  # The maximum may not be available because it may be the broadcast address.
+        return DESTINATION_NODE_ID_MASK  # The maximum may not be available because it may be the broadcast address.
```

Comment:
```suggestion
        return DESTINATION_NODE_ID_MASK
```

The comment no longer holds because the limitations of classless networking do not apply to IP multicasting.